### PR TITLE
Added basic template and logic for rendering report

### PR DESCRIPTION
**Pull request by mariadrywien**
### Link to the Issue

#12 

### Definition of Done

User is able to create a report using parameters (date, commit, test_name, duration_ms) and choosing a desired file path.

### How to test changes
**Help needed**
This task is not finished yet. I created a simple `create_report` function and added its calls in `run_shinytest2_ptest` and `run_cypress_ptest`. Below I described the testing process and my issues.

Create a sample yaml file with params:
```
date:
- 1666173521
- 1666173521
commit:
- 1
- 2
test_name:
- test
- test2
duration_ms:
- 200
- 600
baseline: 1
```

Run the report function with parameters:
```
create_report(report_params = yaml::read_yaml("your_yaml_file.yml", filename = "my_report")
```
An html report will be created in the directory `inst/qmd`. However, in my case, this report is not fully rendered - it doesn't contain plots, tables, etc. When it's rendered, it looks like not all the files are saved.

If we modify the `create_report` function in `R/utils_report.R`, by deleting the `output_file` parameter from `quarto_render`, so:
```
  quarto_render(input = PATH_FILE_REPORT_TEMPLATE, execute_params = report_params)
```
the report is rendered in the `inst/templates` directory and in my case, it's a fully rendered report. I cannot figure out why does specifying the output file break the rendering process. 

Also, I know you wanted to have sth like `shiny.performance::create_report("path_to_report.md")`. We can discuss it as I wasn't sure what is the best approach.

### Tasks for PR author

- [ ] Test your change and ensure there is no regression
- [ ] Change has a corresponding issue. ***Ensure it is linked in GitHub***
- [ ] Author of the change opened a pull request and assigned a reviewer

### General policy:

- If applicable - add instructions for testing
- If there’s no issue, create it. Each issue needs to be well defined and described.
- All interaction with a user, user-facing messages, plots, reports etc. are written from the perspective of the person using or receiving it. They are understandable and helpful to this person. If a user sees an error message, there is a call to action, i.e. the user knows what to do to fix it.
- README, other documentation and code comments that we have is updated with all information related to the change.
- All code has been peer-reviewed before merging into any main branch
- All changes have been merged into the main branch we use for development.
- Continuous integration checks (linter, unit tests, integration tests) are configured and pass.
- Optional: unit tests added for all new or changed logic.
- All task requirements satisfied. If not describe it here. The reviewer is responsible to verify each aspect of the task.
- Change covers only things in task. Please create new PR if you want to fix something else.

### CODE REVIEW

**Comment by mariadrywien** [comment]
I've updated the template based on the structure we were talking about and it works pretty good when I run it manually, but unfortunately there is some error with the yaml file parsing when the report is rendering. I've spent some time on it but couldn't figure it out yet. I hope I'll find some time on Wednesday to fix it, as I wouldn't like to leave it open 😒 Here are steps to reproduce (long 😬 ):
```
# the test output list
test_structure_master <-
  structure(
    list(
      date = structure(
        c(1666173521, 1666173521, 1666173521, 1666173521),
        class = c("POSIXct", "POSIXt"),
        tzone = ""
      ),
      rep_id = c(1L, 1L, 1L, 1L),
      test_name = c(
        "The app starts",
        "Out1 time elapsed",
        "Out2 time elapsed",
        "Out3 time elapsed"
      ),
      duration_ms = c(343L, 6288L, 3118L, 1098L)
    ),
    class = "data.frame",
    row.names = c(NA,-4L)
  )

test_structure_dev <-
  structure(
    list(
      date = structure(
        c(1666969754, 1666969754, 1666969754, 1666969754, 1666969754, 1666969754),
        class = c("POSIXct", "POSIXt"),
        tzone = ""
      ),
      rep_id = c(1L, 1L, 1L, 1L, 1L, 1L),
      test_name = c(
        "The app starts",
        "Out1 time elapsed",
        "Out2 time elapsed",
        "Out3 time elapsed",
        "Out4 time elapsed",
        "Out5 time elapsed"
      ),
      duration_ms = c(743L, 3288L, 1118L, 198L, 3000L, 1000L)
    ),
    class = "data.frame",
    row.names = c(NA,-6L)
  )

out <- list("master" = NULL, "develop" = NULL)
out$master[[1]] <- test_structure_master
out$develop[[1]] <- test_structure_dev
rep_params <- list("test_output" = NULL)
rep_params$test_output <- out
```
Then, when I run: 
```
shiny.benchmark:::create_report(report_params = rep_params, report_name = "whatever")
```
I get:
```
Error in knit_params_get(input_lines, params) : 
  render params not declared in YAML: test_output
Calls: .main -> execute -> <Anonymous> -> knit_params_get
Execution halted
Error in `processx::run(quarto_bin, args, echo = TRUE)`:
! System command 'quarto' failed
---
Exit status: 1
stdout & stderr: <printed>
---
```
Checked how the params are saved to yaml inside `quarto_render` function and it looks fine to me...
image
image
Maybe something will come to my mind during the weekend 🤞

**Comment by DouglasMesquita** [comment]
About the github action errors: It is related to the `devtools::check` that has two notes
- `echarts4r` is in DESCRIPTION but not really in use. Just put it in Suggests instead of Imports (the same for `DT` )
- we are using `system.file` function. I saw you added the `StagedInstall: no` in the description file. I have no idea how to fix it so far but this is important for us to be able to run the tests using github actions

Another comments:

Everything is working fine and we have a default report <o/
I would still add some things (it can be new issues in the project for the next implementations):
- Instead of creating and saving the report in `performance_report` folder, the user should be allowed to select the project path
- It would be good to add a message like: "You can edit this report and re-render it in path/to/file.qmd"
- Maybe another message saying it is experimental? Since we do not have tests for the report. What do you think?

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
-# 0.1.1
+# shiny.benchmark 0.1.1
 
 Adding a minimal example using `shinytest2`
+
+# shiny.benchmark 0.1.9000
+
+Create reports automatically using `report_file` argument in `benchmark` function
```
Comment (jakubnowicki) [inline NEWS.md:7]:
```suggestion
# shiny.benchmark (development)

Create reports automatically using `report_file` argument in `benchmark` function.

# shiny.benchmark 0.1.1

Adding a minimal example using `shinytest2`
```

File: R/utils_report.R [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+#' @title Create a performance report for the tests that were run
+#'
+#' @param report_params list of tests containing commits with dates, duration
+#' time and test name
+#' @param file name of the file to which the report should be saved (.html)
+#'
+#' @return None. This function is called for side effects
+#'
```
Comment (jakubnowicki) [inline R/utils_report.R:48]: Misses examples tag.

**Review by jakubnowicki — APPROVED** [review]
LGTM, please check my comments regarding some minor issues

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Review by DouglasMesquita — APPROVED** [review]

**Pull request merged**